private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,162 @@ func writeConfig(
 	for _, option := range options {
 		option(writeConfigOptions)
 	}
-	if !writeConfigOptions.documentationComments && writeConfigOptions.uncomment {
-		return errors.New("cannot set uncomment without documentationComments for WriteConfig")
+	if err := validateWriteConfigOptions(writeConfigOptions); err != nil {
+		return err
 	}
-	if writeConfigOptions.moduleIdentity == nil && len(writeConfigOptions.dependencyModuleReferences) > 0 {
-		return errors.New("cannot set deps without a name for WriteConfig")
+	// This is the same default as the bufconfig getters.
+	version := V1Beta1Version
+	if writeConfigOptions.version != "" {
+		if err := validateVersion(writeConfigOptions.version); err != nil {
+			return err
+		}
+		version = writeConfigOptions.version
 	}
```

Comment:
Is this right? The default version is `v1` when we run `buf config init`, and the `WriteConfigWithVersion` option says `v1` is still the default, but this looks like `v1beta1` is.

I've tested that `buf config init` continues to write a configuration with `v1` on this branch, so I guess it's working as expected, but I wonder if it's just because we're always providing the options at each call-site.

Reply:
I see now, it's probably right that we should default to `v1beta1` since that's consistent with the configuration getters - we just need to update the documentation on `WriteConfigWithVersion` to say that the default is `v1beta`.